### PR TITLE
[IMP] l10n_in_edi_ewaybill: transporter id must have GST number

### PR DESCRIPTION
**Pull request by zepa-odoo**
In this commit, when user has created a transporter from the invoice view and if it's forget to enter GSTIN
Then we throws error to set GSTIN in transporter

transporter always have GSTIN, without having GSTIN any one is not allowed to transport goods in India. 

task - 3262215

### CODE REVIEW

File: addons/l10n_in_edi_ewaybill/models/account_edi_format.py [hunk 1]
```diff
@@ -77,6 +77,8 @@ def _check_move_configuration(self, move):
             error_message.append(_("- Transportation Mode"))
         elif move.l10n_in_mode == "0" and not move.l10n_in_transporter_id:
             error_message.append(_("- Transporter is required when E-waybill is managed by transporter"))
+        elif move.l10n_in_mode == "0" and move.l10n_in_transporter_id and not move.l10n_in_transporter_id.vat:
+            error_message.append(_("- Selected Transporter not have GSTIN"))
```
Comment (aliyatastemirova) [inline addons/l10n_in_edi_ewaybill/models/account_edi_format.py:81]: How about `- Selected Transporter is missing GSTIN`?

**Review by aliyatastemirova — COMMENTED** [review]
Thanks, I just have one suggestion about the wording. And a very small thing: the convention would be to write `task-3262215` in the commit message :slightly_smiling_face:

File: addons/l10n_in_edi_ewaybill/models/account_edi_format.py (hunk 1 above)
Comment (jva-odoo) [reply addons/l10n_in_edi_ewaybill/models/account_edi_format.py:81]: nice one

Comment (zepa-odoo) [reply addons/l10n_in_edi_ewaybill/models/account_edi_format.py]: I made some changes to the pull request based on your feedback.

File: addons/l10n_in_edi_ewaybill/models/account_edi_format.py [hunk 2]
```diff
@@ -77,6 +77,8 @@ def _check_move_configuration(self, move):
             error_message.append(_("- Transportation Mode"))
         elif move.l10n_in_mode == "0" and not move.l10n_in_transporter_id:
             error_message.append(_("- Transporter is required when E-waybill is managed by transporter"))
+        elif move.l10n_in_mode == "0" and move.l10n_in_transporter_id and not move.l10n_in_transporter_id.vat:
+            error_message.append(_("- Selected Transporter is missing GSTIN?"))
```
Comment (aliyatastemirova) [inline addons/l10n_in_edi_ewaybill/models/account_edi_format.py:81]:
```suggestion
            error_message.append(_("- Selected Transporter is missing GSTIN"))
```

**Review by aliyatastemirova — CHANGES_REQUESTED** [review]

File: addons/l10n_in_edi_ewaybill/models/account_edi_format.py (hunk 2 above)
Comment (aliyatastemirova) [reply addons/l10n_in_edi_ewaybill/models/account_edi_format.py:81]: Also, we should add it in translation files :slightly_smiling_face:

Comment (zepa-odoo) [reply addons/l10n_in_edi_ewaybill/models/account_edi_format.py]: I made some changes to the pull request based on your feedback.
Can you re-check this?

Thank you

**Review by jva-odoo — APPROVED** [review]

**Pull request closed without merging**